sphinx_needs/api/need.py

```diff
@@ -412,16 +412,31 @@ def run():
 
     node_need_content = _render_template(content, docname, lineno, state)
 
-    # Extract first plantuml diagram
+    # Extract plantuml diagrams
+    node_need_needumls_without_key = []
+    node_need_needumls_key_names = []
     for child in node_need_content.children:
         if isinstance(child, Needuml):
             needuml_id = child.rawsource
             try:
                 needuml = env.needs_all_needumls.get(needuml_id)
-                needs_info["diagram"] = needuml["content"]
+                if needuml["key"]:
+                    key_name = needuml["key"]
+                    if key_name not in node_need_needumls_key_names:
```

Comment:
MAybe we should check, if `key` is already set in the need or if it is defined in needs_extra_options or needs_extra_links.

In all cases, this is an error, as we overwrite some stuff.